### PR TITLE
Metavalue number of datapoints

### PR DESCRIPTION
**Pull request by vrossow**
## Description

New MetaValue number_of_datapoints, represents the number of datapoints used to calculate the intensity of a feature. It is meant to be similar to MaxQuants number_of_datapoints.
Will be used in a future version of the MQEvidenceExporter.

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/include/OpenMS/CONCEPT/Constants.h [hunk 1]
```diff
@@ -482,6 +482,12 @@ namespace OpenMS
               String
       */
       inline const std::string   NUM_OF_MASSTRACES = "num_of_masstraces";
+
+      /** User parameter name for the number of datapoints ()like in MaxQuant in a feature. (Required for MQEvidenceExporter)
```
Comment (cbielow) [inline src/openms/include/OpenMS/CONCEPT/Constants.h]:
```suggestion
      /** User parameter name for the total number of data points (peaks) in a feature. (Required for MQEvidenceExporter)
```

File: src/tests/topp/FeatureFinderCentroided_1_1_output.featureXML [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+				<pt x="4175.729999999999564" y="649.7698974609375" />
+				<pt x="4218.890000000000327" y="649.7698974609375" />
+				<pt x="4218.890000000000327" y="649.7877197265625" />
+				<pt x="4175.729999999999564" y="649.7877197265625" />
+			</convexhull>
+			<UserParam type="int" name="label" value="7"/>
+			<UserParam type="float" name="score_fit" value="0.876456086169051"/>
+			<UserParam type="float" name="score_correlation" value="0.975213375865844"/>
+			<UserParam type="float" name="FWHM" value="34.366207122802734"/>
+			<UserParam type="int" name="number_of_datapoints" value="47"/>
```
Comment (cbielow) [inline src/tests/topp/FeatureFinderCentroided_1_1_output.featureXML]: did you check using TOPPView? Just to make sure...

File: src/tests/topp/CMakeLists.txt [hunk 3]
```diff
@@ -286,7 +286,7 @@ set_tests_properties("TOPP_FeatureFinderMetabo_5_out" PROPERTIES DEPENDS "TOPP_F
 #------------------------------------------------------------------------------
 # FeatureFinderCentroided test
 add_test("TOPP_FeatureFinderCentroided_1" ${TOPP_BIN_PATH}/FeatureFinderCentroided -test -ini ${DATA_DIR_TOPP}/FeatureFinderCentroided_1_parameters.ini -in ${DATA_DIR_TOPP}/FeatureFinderCentroided_1_input.mzML -out FeatureFinderCentroided_1.tmp)
-add_test("TOPP_FeatureFinderCentroided_1_out1" ${DIFF} -whitelist "id=" -in1 FeatureFinderCentroided_1.tmp -in2 ${DATA_DIR_TOPP}/FeatureFinderCentroided_1_output.featureXML )
+add_test("TOPP_FeatureFinderCentroided_1_out1" ${DIFF} -whitelist "id=" -in1 FeatureFinderCentroided_1.tmp -in2 ${DATA_DIR_TOPP}/FeatureFinderCentroided_1_1_output.featureXML )
```
Comment (cbielow) [inline src/tests/topp/CMakeLists.txt:289]: why not simply replace the old file?
AFAIK this output file is only used for this test.
Adding a new one would orphan `FeatureFinderCentroided_1_output.featureXML`?

Comment (vrossow) [reply src/tests/topp/CMakeLists.txt:289]: Actually wanted to ask that later, if we should delete/overwrite the old one. But I guess I have the answer now.

**Comment by cbielow** [comment]
maybe add a test for the metavalue here?
https://github.com/cbielow/OpenMS/blob/develop/src/tests/class_tests/openms/source/FeatureFinderAlgorithmPicked_test.cpp#L100

**Comment by cbielow** [comment]
closing since merged into OpenMS already

**Pull request closed without merging**